### ISSUE TITLE
Feature Wishlist

### ISSUE THREAD

**Issue by raman325** [home-assistant-libs/zwave-js-server-python#364]
Opening a new issue in case anyone has a feature they want/need that's not on the roadmap. This is open to all but not a support avenue. Additionally, if you like an idea, add a +1 and do not add another comment.

**Comment by raman325** [comment]
> I fear it's out of scope, but I would _love_ a non-async interface to this. I would love to be able to pass something a websocket URL, and get a bunch of objects that represent the controller and nodes with all of the async updating stuff going on behind the scenes.

Yes this is out of scope, assuming it's even possible it would have to be rebuilt from the ground up, likely using multiple threads. Out of curiosity, what's your use case?

**Comment by raman325** [comment]
Yeah the model works best when the client is long lived so that it can receive smaller state dumps after the initial dump. Separately, I am going to mark all of these comments as off topic to keep the request list clean

**Comment by bronger** [comment]
A complete, non-trivial yet short example would be helpful in my opinion.

**Comment by raman325** [comment]
> A complete, non-trivial yet short example would be helpful in my opinion.

What kind of example are you interested in?

**Comment by bronger** [comment]
A minimal working program that e.g. polls one value of a certain device. Of course, the code would not run anywhere, but one could see the core structure that a program using this library must have.

**Comment by raman325** [comment]
gotcha, we'll take a working example from someone who is willing to contribute it but in the interim the zwave_js integration in Home Assistant is the best example. I recognize that if you aren't familiar with the HA code, it may be hard to parse: https://github.com/home-assistant/core/tree/dev/homeassistant/components/zwave_js but start with the `Client` and work your way from there
